articles/coding-on-purpose.md

```diff
@@ -0,0 +1,99 @@
+* Code, which is mostly imperative in it's nature, does just declare *how* something is achieved but it does not state *why*. So while the purpose was hopefully clear to the one writing the code, it is likely to be unknown to the poor coder maintaining the code years later.
+* I've seen too many user stories that either don't state a goal and benefit or if they do, it's just a wild guess by somebody who wanted to satisfy the template structure. And worse: higher level items like *epics* miss the *purpose* even more often.
+* BDD (Behaviour Driven Design) typically uses the *Given, When, Then* pattern which only describes what the software does, but not why.
+* Many consulting companies get told *what* to build instead of *what problem to solve* or *what outcome to achieve*.
+* Inexperienced managers might tell you *what* to build instead of *what outcome to achieve*.
+
+## Everyday Tips
+
+### When The Solution Becomes The Problem
+
+Sometimes you're stuck deep down in a tricky problem. While some people might just give up, you're passion for problem solving lets you breed over the puzzle for hours and days, maybe even weeks. While working on the solution for your customers problem, the solution suddenly turned into the problem. Or as [Paul Watzlawick says](https://www.youtube.com/watch?v=7etsh4HwG78):
+
+> By searching for solutions we restrict ourselves with constraints which don't exist in the original problem.
+
+When you're stuck in such a situation it's worth taking one or two steps back and asking: *what again was the goal we wanted to achieve?* And then again: *why did we want to achieve this goal in the first place*? This helps you opening up the space for possible solutions and hopefully let's you discover that the problem you're trying to solve does not even need solving but instead you find a simpler, less problematic solution to the initial *why*. If you ask *why* five times, this approach even has a name: [The five whys technique](https://en.wikipedia.org/wiki/5_Whys).
+
+### Measure Outcome, Not Output
+
+Think about how you track achievement of desired outcome in your project. In the end, it's the job of your *Product Owner* or *Product Manager*, but if you think they could use some inspiration, they are probably greatful for some ideas. As an example, [John Cutler proposes](https://twitter.com/johncutlefish/status/775523444385382400) to add a column to your task board called *Achieved desired outcome*. Or [Gregor Ilg says](https://twitter.com/_gregorilg/status/884663883172110336):
+
+> Don't celebrate when you launched a product. Celebrate once you've learned from it.
+
+Instead of measuring how many story points we can deliver per sprint, we should apply our math skills much more to measuring and tracking whether we achieve the desired outcomes in production. Do we get the desired user adoption? Is the desired percentage of business done on the new platform? Do we save the desired amount of manual work using the new system?
+

```

Comment:
This one is the best (Measure Outcome, Not Output)! I would call it "Measure Impact, Not Output", because "Outcome" is almost as vague as "Output", isnt it?